guillotina/renderers.py

```diff
@@ -8,33 +8,35 @@
 from typing import Optional
 from zope.interface.interface import InterfaceClass
 
-import json
-import ujson
-
-
-class GuillotinaJSONEncoder(json.JSONEncoder):
-    def default(self, obj):
-        if isinstance(obj, complex):
-            return [obj.real, obj.imag]
-        elif isinstance(obj, datetime):
-            return obj.isoformat()
-        elif isinstance(obj, type):
-            return obj.__module__ + "." + obj.__name__
-        elif isinstance(obj, InterfaceClass):
-            return [x.__module__ + "." + x.__name__ for x in obj.__iro__]  # noqa
-        try:
-            iterable = iter(obj)
-        except TypeError:
-            pass
-        else:
-            return list(iterable)
-
-        if isinstance(obj, PermissionSetting):
-            return obj.get_name()
-        if callable(obj):
-            return obj.__module__ + "." + obj.__name__
-        # Let the base class default method raise the TypeError
-        return json.JSONEncoder.default(self, obj)
+import orjson
+
+
+def guillotina_json_default(obj):
+    if isinstance(obj, str):
+        if type(obj) != str:  # e.g, i18n.Message()
+            return str(obj)
+    if isinstance(obj, complex):
+        return [obj.real, obj.imag]
+    elif isinstance(obj, datetime):
+        return obj.isoformat()
+    elif isinstance(obj, type):
+        return obj.__module__ + "." + obj.__name__
+    elif isinstance(obj, InterfaceClass):
+        return [x.__module__ + "." + x.__name__ for x in obj.__iro__]  # noqa
+    try:
+        iterable = iter(obj)
```

Comment:
this also seems like not so optimus...

Reply:
I'm not sure what's the purpose of this code. It seems this is here from Guillotina 1.0.0 (sept 2016). Does anyone know if this is still needed? cc/ @bloodbare @vangheem

Reply:
AFAIR its for iterables like a Set, json and ujson were not able to convert Set objects.

Reply:
Also for something like dict.keys() You get and inter object not a list